dns-rc-autoscaler/nanny/scaler.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nanny
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// Scaler determines the number of replicas to run
+type Scaler struct {
+	ConfigFile string
+	Verbose    bool
+}
+
+const paramsSeparator = "nodes.gte."
+
+func (s Scaler) scaleWithNodes(numCurrentNodes uint64) uint64 {
+	newMap, err := ParseScalerParamsFile(s.ConfigFile)
+	if err != nil {
+		log.Fatalf("Parse failure: The configmap volume file is malformed (%s)\n", err)
+	}
+
+	// construct a search ladder from the map
```

Comment:
As much as I love optimiziation, this is small-n, so linear will almost
always be faster.

On Tue, Aug 2, 2016 at 1:40 PM, Q-Lee <notifications@github.com> wrote:

> In dns-rc-autoscaler/nanny/scaler.go
> <https://github.com/kubernetes/contrib/pull/1427#discussion_r73232734>:
>
> > +
> > +// Scaler determines the number of replicas to run
> > +type Scaler struct {
> > +	ConfigFile string
> > +	Verbose    bool
> > +}
> > +
> > +const paramsSeparator = "nodes.gte."
> > +
> > +func (s Scaler) scaleWithNodes(numCurrentNodes uint64) uint64 {
> > +	newMap, err := ParseScalerParamsFile(s.ConfigFile)
> > +	if err != nil {
> > +		log.Fatalf("Parse failure: The configmap volume file is malformed (%s)\n", err)
> > +	}
> > +
> > +	// construct a search ladder from the map
>
> If you do linear search, then just use the map. If you're going to sort,
> then do binary search. If you spend O(nlogn) building a searchable data
> structure, then do it during initialization and amortize over multiple
> queries.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1427/files/0bc392d4d1d2508aea02e27c3a8a6c34fe5e4a4f#r73232734>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVHopcTG4_v7bvcp5fAI9loTjyDkUks5qb6shgaJpZM4JTRpr>
> .
>